### DISCUSSION TITLE
what is the difference between a pipeline connector mysql and a source connecor mysql cdc?

### DISCUSSION THREAD

**Discussion by plum7ree** [apache/flink-cdc#3803] [Q&A] [answered]
Hello, thanks for the nice software.

I am trying to use a flink cdc and have found out there are two mysql connectors.

pipeline: https://nightlies.apache.org/flink/flink-cdc-docs-master/docs/connectors/pipeline-connectors/mysql/
source: https://nightlies.apache.org/flink/flink-cdc-docs-master/docs/connectors/flink-sources/mysql-cdc/

I had thought that a pipeline mysql is only for a snapshot, 
but when I put the data into mysql, my kafka-sink got real time updated data.

So now I am very confused, what's the difference between those. 

Can anyone explain what's the purpose of those two? 
Does pipeline mysql connector also doing CDC job (there are option debezium on yaml file)?
Then why there is also a source mysql-cdc connector?

**Comment by gong** [comment, marked as answer]
@plum7ree pipeline connector mysql is dependent on source connector mysql. pipeline connector is cdc3.0 for supporting upstream and downstream etl. source connector mysql is just cdc source.

**Reply by plum7ree** [reply]
So, the pipeline connector also supports CDC . Thanks !

**Reply by zero2hj** [reply]
> @plum7ree pipeline connector mysql is dependent on source connector mysql. pipeline connector is cdc3.0 for supporting upstream and downstream etl. source connector mysql is just cdc source.
@gong 
but how to understand that flink-cdc-pipeline-mysql connector can only use as source of pipeline which described on the [doc](https://nightlies.apache.org/flink/flink-cdc-docs-master/docs/connectors/pipeline-connectors/overview/), what do you mean upstream and downstream etl here?

**Reply by gong** [reply]
> > @plum7ree pipeline connector mysql is dependent on source connector mysql. pipeline connector is cdc3.0 for supporting upstream and downstream etl. source connector mysql is just cdc source.
> > @gong
> > but how to understand that flink-cdc-pipeline-mysql connector can only use as source of pipeline which described on the [doc](https://nightlies.apache.org/flink/flink-cdc-docs-master/docs/connectors/pipeline-connectors/overview/), what do you mean upstream and downstream etl here?

@zero2hj pipeline include upstream and downstream, flink-cdc-pipeline-mysql connector is upstream.
